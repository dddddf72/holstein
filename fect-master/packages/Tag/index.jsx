import { computed, toRefs, ref, watchEffect } from 'vue'
import { createNameSpace, theme, validator } from '../utils'
import './tag.less'

const [createComponent] = createNameSpace('Tag')
const { normalTypes } = theme

const queryColors = (type, invert) => {
  const colors = {
    default: {
      color: 'var(--primary-foreground)',
    },
    success: {
      color: 'var(--success-default)',
    },
    warning: {
      color: 'var(--warning-default)',
    },
    error: {
      color: 'var(--error-default)',
    },
  }

  const hideBorder = invert

  const invertColor = (hideBorder) => {
    if (hideBorder) {
      const style = {
        color: 'var(--primary-background)',
      }
      return style
    }
    return colors[type]
  }

  /**
   * use hideBorder to control backgroundColor and borderColor
   * when hideBorder value as true .It means should set default style Or it means should
   * set invert style.
   * 
   */
  const tagStyle = {
    ...invertColor(hideBorder),
    bgColor: hideBorder ? colors[type].color : 'var(--primary-background)',
    borderColor: hideBorder ? 'transparent' : colors[type].color,
  }

  return tagStyle
}

export default createComponent({
  props: {
    text: {
      type: String,
      default: '',
    },
    type: {
      type: String,
      default: 'default',
      validator: validator.enums(normalTypes),
    },
    useInvert: Boolean,
  },
  setup(props, { attrs, slots, emit }) {
    const setTagStyle = computed(() => {
      const { type, useInvert } = props
      const { color, bgColor, borderColor } = queryColors(type, useInvert)

      const style = {
        borderColor,
        color: color,
        backgroundColor: bgColor,
      }
      return style
    })

    return () => (
      <div class="fect-tag" style={setTagStyle.value}>
        {props.text}
      </div>
    )
  },
})
