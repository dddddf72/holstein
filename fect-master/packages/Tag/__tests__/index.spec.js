import Tag from '../index'
import { mount } from '@vue/test-utils'

describe('Tag', () => {
  it('should be render as element', () => {
    const wrapper = mount(Tag, {
      props: {
        text: 'Test',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should support different type', () => {
    const wrapper = mount({
      setup() {
        return {
          types: ['default', 'success', 'warning', 'error'],
        }
      },
      render() {
        return (
          <>
            {this.types.map((type, i) => (
              <Tag text={i} type={type} key={type} />
            ))}
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should support invert ', () => {
    const wrapper = mount({
      setup() {
        return {
          types: ['default', 'success', 'warning', 'error'],
        }
      },
      render() {
        return (
          <>
            {this.types.map((type, i) => (
              <Tag text={i} type={type} key={type} useInvert />
            ))}
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
