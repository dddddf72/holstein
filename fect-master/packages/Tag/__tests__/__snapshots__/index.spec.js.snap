// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should be render as element 1`] = `<div class="fect-tag">Test</div>`;

exports[`Tag should support different type 1`] = `
<div class="fect-tag">0</div>
<div class="fect-tag">1</div>
<div class="fect-tag">2</div>
<div class="fect-tag">3</div>
`;

exports[`Tag should support invert  1`] = `
<div class="fect-tag" style="border-color: transparent;">0</div>
<div class="fect-tag" style="border-color: transparent;">1</div>
<div class="fect-tag" style="border-color: transparent;">2</div>
<div class="fect-tag" style="border-color: transparent;">3</div>
`;
